proto/stream_plan.proto

```diff
@@ -259,6 +259,10 @@ message ChainNode {
   bool disable_rearrange = 4;
   // Whether to place this chain on the same worker node as upstream actors.
   bool same_worker_node = 5;
+  // Whether the upstream materialize is and this chain should be a singleton.
+  // FIXME: This is a workaround for fragmenter since the distribution info will be lost if there's only one
+  // fragment in the downstream mview. Remove this when we refactor the fragmenter.
+  bool is_singleton = 6;
```

Comment:
What is the desired way?

Reply:
We may directly operate on the `Fragment` struct in the frontend instead of the protobuf struct, so that we may be able to directly find whether a fragment is a singleton or not instead of checking the dispatcher type.

Currently, we cannot check whether `is_singleton` for the most upstream fragment since there's no exchange before it. That's why we need hacking for `TopN` and `Chain`. 😢